libs/clients/middlewares/src/lib/withMetrics.ts

```diff
@@ -0,0 +1,21 @@
+import { dogStatsD } from '@island.is/infra-metrics'
+
+import { FetchAPI } from './nodeFetch'
+
+export interface MetricsMiddlewareOptions {
+  fetch: FetchAPI
+  name: string
+}
+
+export const withMetrics = ({
+  fetch,
+  name,
+}: MetricsMiddlewareOptions): FetchAPI => {
+  const client = dogStatsD({
+    prefix: `${name}.`,
+  })
+  return (input, init) => {
+    client.increment('requests')
+    return fetch(input, init)
```

Comment:
Would be nice to add a couple more metrics while we're at it:

* ${name}.requests.ok
* ${name}.requests.fail (either separate metrics or tag: `reason=4xx|5xx|network`)

Maybe also a responseTime metric?

The only tricky thing here is that the same API might have quite different endpoints which in some cases we'd want to count separately. If that happens we could expand the Enhanced Fetch API with a callback that customizes the metric keys/tags using request details.

Reply:
I would suggest distinguishing between 4xx and 5xx errors, so have
${name}.requests.ok - {100-399}
${name}.requests.error - {400-499}
${name}.requests.fail - {500 - 599}

Reply:
I added more descriptive error metrics, they are little bit different than suggested. I followed the pattern I was seeing on existing metrics in datadog.

The responseTime metric does not seem to be trivial and the other thing can be an improvement later on.

Just to be clear this PR is a simple example on how to add metric collecting so we have an example to follow.